draft-dekater-scion-dataplane.md

```diff
@@ -1353,6 +1353,16 @@ This section describes the steps that a SCION egress border router MUST perform
 **Note:** For more information on the path meta header, see [](#PathMetaHdr).
 
 
+#### Effects of Clock Inaccuracy
+
+Each router along a packet's path verifies the validity of the current hop field by comparing the current time with the hop's expiration time.
+
+This expiration time is calculated as described in [](#hopfld) on the basis of the segment's timestamp. That timestamp is assigned by the host that originates the segment. A fast clock at origination or a slow clock at a router will yield a lengthened time-to-live; possibly an origination time in the future. A slow clock at origination or a fast clock at a router will yield a shortened time to live; possibly an expiration time in the past.
+
+The shortest time-to-live is 5 minutes, 37 seconds, and 500 milliseconds ([](#hopfld)) and a segment is valid up to that same amount of time prior to its origination timestamp ([](#inffield)). Assuming segments are originated at least once a minute, if the clock difference between the originator of a path and any routers that it refers to does not exceeds 4 minutes and 37 seconds, the shortest-lived segment and the most recent segment are usable. To that end, a clock accuracy better than 1 minute is more than sufficient.
```

Comment:
"Assuming segments are originated at least once a minute ..."

This is not entirely accurate. Longer paths take longer to create. With a 1 minute beaconing period, constructing a path segment that traverses N ASes will take up to N minutes.

With this in mind, I don't really know how to write this. Perhaps something to the effect of: the expiration time of a path can often not be constrained too tightly, as the path discovery takes a variable amount of time and the usable lifetime of the path is  reduced accordingly. Compared to this effect, even clock drifts in the order of minutes do not pose additional problems.

Reply:
Ah yes, that's even more relevant...unfortunately not as clear-cut indeed.